tap_dynamodb/sync.py

```diff
@@ -25,6 +26,16 @@ def sync_stream(config, state, stream):
     table_name = stream['tap_stream_id']
 
     md_map = metadata.to_map(stream['metadata'])
+
+    # added this extra check here to catch the JSONDecodeError early.
+    try:
+        expression = metadata.get(md_map, (), 'tap-dynamodb.expression')
+        # to verify `json.loads()` is only called if expression is not None or empty
+        if expression:
+            json.loads(expression)
```

Comment:
@namrata270998 It is better to add this try except block in a function and use it at the required places

Reply:
Updated the code as per suggestions